src/gwt/panmirror/src/editor/src/optional/ace/ace.ts

```diff
@@ -455,6 +501,16 @@ export class AceNodeView implements NodeView {
       }
     });
 
+    // If the cursor moves and we're in focus, ensure that the cursor is
+    // visible. Ace's own cursor visiblity mechanisms don't work in embedded
+    // editors.
+    this.editSession.getSelection().on('changeCursor', () => {
```

Comment:
Is listening for `beforeEndOperation` signals from the editor instance itself an option here? This is basically where Ace itself looks and tries to scroll into view after a command is executed:

https://github.com/ajaxorg/ace/blob/3ebb0c34dc023aff3b6a951dd116a702510369fe/lib/ace/editor.js#L175

Reply:
I'll look into that. The first couple tries at this fix were unsuccessful because scrolling the document immediately after the cursor moved generated jumpiness and unwanted selections (Ace wasn't really done).

Reply:
@kevinushey `beforeEndOperation` is indeed better! Updated.

Reply:
Awesome; LGTM!